### DISCUSSION TITLE
How can I password protect my site?

### DISCUSSION THREAD

**Discussion by jamiecropley** [vercel/community#4226] [Feature Question] [answered]
### Question

$150 a month to password protect one site? Just checking, is this a typo? Seems a bit steep. This is per deployment it seems as well per month, so say I have 20 sites I wanted to do this on thats $36000 for one feature. Even on a backend I don't understand the cost here, not complaining, it just seems a bit steep and technically I don't understand why, thus I wonder if it could be a typo meant to indicate $15 a month? What do you reckon? Looking at competitors like netlify as well they include such feature in the price.

![image](https://github.com/vercel/community/assets/17186004/3a39d20e-7dfb-4eb2-aadf-55695203ad87)

**Comment by leerob** [comment, marked as answer]
Hey there, if you would prefer not to use this (definitely heard), then you can use [Deployment Protection](https://vercel.com/docs/security/deployment-protection) for free. Alternatively, you can create a basic email/password auth using this [template](https://vercel.com/templates/next.js/basic-auth-password) with Middleware.

**Reply by jamiecropley** [reply]
Thanks for the help! I tend not to look through the options usually as they confuse me haha.

So for example: 
Pro | 1 million | $0.65 per 1 million invocations

Means I would have to reach 1 million logins within the space of 'one' month until I am charged, where if its just me logging in then its no problem?

**Reply by leerob** [reply]
1 million Edge Middleware invocations per month are included in the Pro plan, correct. Then another 1 million would cost you $0.65.

**Reply by jamiecropley** [reply]
ok thanks

**Reply by holmey** [reply]
I faced a similar challenge. I actually only needed a simple Basic Authentication for a completely static HTML page. Since I'm not using Next.js, the middleware solution, in my view, isn't an option.

I stumbled upon this discussion: https://github.com/orgs/vercel/discussions/2920. I gave it a try and as a result, I now have a simple Basic Authentication.

I think this example could also be added to the documentation. Often, all you need is straightforward password protection. Or am I using the `vercel.json` incorrectly here @leerob?
